manager/integration/tests/test_node.py

```diff
@@ -2239,4 +2256,282 @@ def test_node_eviction():
     the same as checksum 1 and checksum 2.
     28. Set 'Eviction Requested' to 'false' and enable scheduling on node 2.
     """
-    pass
+    nodes = client.list_node()
+
+    node1 = nodes[0]
+    node2 = nodes[1]
+    node3 = nodes[2]
+    client.update(node3, allowScheduling=False)
+
+    volume1_name = volume_name + "-1"
+    volume1_size = str(500 * Mi)
+    volume1_data_path = "/data/test"
+    pv1_name = volume1_name + "-pv"
+    pvc1_name = volume1_name + "-pvc"
+    pod1_name = volume1_name + "-pod"
+    pod1 = copy.deepcopy(pod)
+
+    pod1['metadata']['name'] = pod1_name
+    pod1['spec']['volumes'] = [{
+        "name": "pod-data",
+        "persistentVolumeClaim": {
+            "claimName": pvc1_name
+        }
+    }]
+
+    pod1['spec']['nodeSelector'] = \
+        {"kubernetes.io/hostname": node3.name}
+
+    volume1 = create_and_check_volume(client,
+                                      volume1_name,
+                                      num_of_replicas=2,
+                                      size=volume1_size)
+
+    create_pv_for_volume(client, core_api, volume1, pv1_name)
+    create_pvc_for_volume(client, core_api, volume1, pvc1_name)
+
+    create_and_wait_pod(core_api, pod1)
+    volume1 = wait_for_volume_healthy(client, volume1_name)
+
+    volume1_replica1 = volume1.replicas[0]
+    volume1_replica2 = volume1.replicas[1]
+
+    assert volume1_replica1.mode == "RW"
+    assert volume1_replica2.mode == "RW"
+    assert volume1_replica1.running is True
+    assert volume1_replica2.running is True
+
+    write_pod_volume_random_data(core_api,
+                                 pod1_name,
+                                 volume1_data_path,
+                                 DATA_SIZE_IN_MB_2)
+
+    volume1_md5sum = get_pod_data_md5sum(core_api,
+                                         pod1_name,
+                                         volume1_data_path)
+
+    client.update(node1, allowScheduling=False, evictionRequested=True)
+    wait_for_volume_replica_count(client, volume1_name, 3)
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    volume1_err_replica = None
+    for replica in volume1.replicas:
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
+        else:
+            volume1_err_replica = replica
+            break
+    assert volume1_err_replica is not None
+    assert volume1_err_replica.running is False
+    assert volume1_err_replica.mode == ''
+
+    client.update(node1, allowScheduling=False, evictionRequested=False)
+    wait_for_volume_replica_count(client, volume1_name, 2)
+
+    volume1 = client.by_id_volume(volume1_name)
+
+    for replica in volume1.replicas:
+        if replica.name == volume1_replica1.name or \
+           replica.name == volume1_replica2.name:
+            assert replica.running is True
+            assert replica.mode == "RW"
```

Comment:
Can we also check if one replica is on node-1, one is on node-2?  Since for this test the location of the replicas are very important. Thanks.

Reply:
fixed